homeassistant/components/tplink/config_flow.py

```diff
@@ -101,14 +214,65 @@ async def async_step_user(
             errors=errors,
         )
 
+    async def async_step_user_auth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Dialog that informs the user that auth is required."""
+        errors = {}
+        host = self.context[CONF_HOST]
+        assert self._discovered_device is not None
+        if user_input:
+            credentials = Credentials(
```

Comment:
I'd unpack username and password into local vars since you are likely to do a dict lookup below as well

Reply:
Done